File: R/8kyu/8kyu-reversed-words.r
# Reversed Words
# Language: R
# Version: 3.4.1
# Difficulty: 8 kyu

solution <- function(sentence){
  paste(rev(unlist(strsplit(sentence, " "))), collapse = " ")
}


# Tests
test_that('Basic tests', {
  expect_equal(solution("hello world!"), "world! hello")
  expect_equal(solution("yoda doesn't speak like this"), "this like speak doesn't yoda")
  expect_equal(solution("foobar"), "foobar")
  expect_equal(solution("kata editor"), "editor kata")
  expect_equal(solution("row row row your boat"), "boat your row row row")
})

File: Python/8kyu/8kyu-reversed-words.py
# Reversed Words
# Language: Python
# Version: 3.10
# Difficulty: 8 kyu

# Solution 1
def reverse_words(s):
    return " ".join(reversed(s.split(" ")))

# Refactored Solution 1
def reverseWords(s):
    return " ".join(s.split(" ")[::-1])


# Tests
import codewars_test as test

try:
    from solution import reverseWords as reverse_words
except ImportError:
    from solution import reverse_words

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it("Basic Tests")
    def basic_tests():
        test.assert_equals(reverse_words("hello world!"), "world! hello")